### DISCUSSION TITLE
Simulating the state drift cuased by low voltages applied on the PCM devices during inferences

### DISCUSSION THREAD

**Discussion by cdm233** [IBM/aihwkit#458] [Q&A]
Dear all, 

I see in the document that it is possible to simulate the intrinsic drift of the devices after a set period of time, but is it possible to simulate the state drift caused by applying low voltages to the PCM devices during inference? To put it concretely: is it possible to simulate the weight change after n inferences where during each inference we apply a pulse of V_i magnitude for P_i nanoseconds to the crossbar circuit? 

This paper may help to understand my question: https://ieeexplore.ieee.org/abstract/document/6800398

Many thanks

**Comment by maljoras** [comment]
Hi @cdm233 ,
many thanks for the question. For inference with PCM, we apply a static noise / drift distribution to the weights which is time dependent. This noise / drift distribution is estimated from PCM measurements. So we do not in detail model the temporal evolution of each individual weight for a given time after programming. In your case a time dependent evolution of the weight (conductance) value would be needed, so the toolkit would not support your experiment. You would have to write your own temporal evolution of the weights and set the weight matrix to a given value at time t and then run an inference with that weight matrix (assuming that one inference is fast enough that the weight matrix does not change significantly).
